src/utils.ts

```diff
@@ -34,12 +36,15 @@ export function optionFromList(option: string, values: string[]): string[] {
     return values.map((value) => [option, value]).flat();
 }
 
-export function hashFilesOrEmpty(patterns: string[]): string {
-    try {
-        return hashFiles.sync({ files: patterns });
-    } catch (error) {
-        return "";
-    }
+export async function hashFolderContent(path: string): Promise<string> {
+    return (await hashElement(path, { encoding: "hex", folders: { ignoreRootName: true } })).hash;
+}
+
+export async function hashFiles(patterns: string[] = []): Promise<string> {
+    const hasher = crypto.createHash("md5");
+    const nodes = (await glob(patterns)).sort().map((filename) => hashElement(filename));
+    (await Promise.all(nodes)).forEach((node) => hasher.update(node.hash));
```

Comment:
In both of these cases, can you assign the `await` result to an intermediate variable and then use it? `(await <complex expression>).<stuff>` is not very readable, and `return (await <complex expr>).<stuff>` is even worse.

Reply:
You're right, I hadn't noticed how ugly it was before. I've changed it, could you have a look if it is better now? ;)

Reply:
Sure, will look in a minute!

For situations like this, self-reviewing your PR in the GitHub interface right after you open it is super useful. When you're in the middle of trying to get things working, things like these are easy to miss. But I bet you would have noticed both issues I flagged here by yourself in a self-review — I don't think I contributed anything special there.

Reply:
I always go through the changes I made in the GitHub interface, and this was the case too, but maybe I wasn't careful enough. I'll try to do better next time!